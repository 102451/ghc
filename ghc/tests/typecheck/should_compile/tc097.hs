--!!! Local universal quantification.
module ShouldSucceed where

import PrelGHC -- to get at All

data Monad2 m = MkMonad2 (All a => a -> m a)
                         ((All a, All b) =>  m a -> (a -> m b) -> m b)

halfListMonad  :: ((All a, All b) => [a] -> (a -> [b]) -> [b]) -> Monad2 []
halfListMonad b = MkMonad2 (\x -> [x]) b
