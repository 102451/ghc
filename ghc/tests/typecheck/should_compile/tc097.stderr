__export ShouldSucceed Monad2{MkMonad2} halfListMonad;
1 data Monad2 m :: (* -> *) = MkMonad2 (__forall [a] => a -> m a) (__forall [b a] => m a -> (a -> m b) -> m b) ;
1 halfListMonad :: (__forall [b a] => [a] -> (a -> [b]) -> [b]) -> Monad2 PrelBase.ZMZN ;
